components/BarcodeScanner.vue

```diff
@@ -95,8 +95,10 @@ export default {
       const defaultCameraId =
         backCameras.at(-1)?.deviceId ?? cameraList.value.at(-1)?.deviceId;
       await scanner.start(
-        selectedCameraId.value ?? defaultCameraId,
-        getCameraScanConfig(), onScanSuccess, onError
+        {deviceId: {exact: selectedCameraId.value ?? defaultCameraId}},
+        getCameraScanConfig(),
+        onScanSuccess,
+        onError
       );
```

Comment:
No dropped parens w/o a curly brace friend.

```suggestion
        onError);
```